schematic/schemas/create_json_schema.py

```diff
@@ -201,168 +207,80 @@ def __post_init__(self) -> None:
-
-    return (
-        prop_type,
-        is_array,
-        minimum,
-        maximum,
-        pattern,
-    )
-
-
-def _get_range_from_in_range_rule(
-    rule: str,
-) -> tuple[Optional[float], Optional[float]]:
-    """
-    Returns the min and max from an inRange rule if they exist
-
     Arguments:
-        rule: The inRange rule
+        validation_rules: A list of input validation rules
 
     Returns:
-        The min and max from the rule
-    """
-    range_min: Optional[float] = None
-    range_max: Optional[float] = None
-    parameters = rule.split(" ")
-    if len(parameters) > 1 and parameters[1].isnumeric():
-        range_min = float(parameters[1])
-    if len(parameters) > 2 and parameters[2].isnumeric():
-        range_max = float(parameters[2])
-    return (range_min, range_max)
-
+        A tuple containing fields for a Node object:
+        - js_is_array: Whether or not the Node should be an array in JSON Schema
+        - js_type: The JSON Schema type
+        - js_format: The JSON Schema format
+        - js_minimum: If the type is numeric the JSON Schema minimum
+        - js_maximum: If the type is numeric the JSON Schema maximum
+        - js_pattern: If the type is string the JSON Schema pattern
+    """
+    js_is_array = False
+    js_type: Optional[str] = None
+    js_format: Optional[str] = None
+    js_minimum: Optional[float] = None
+    js_maximum: Optional[float] = None
+    js_pattern: Optional[str] = None
 
-def _get_pattern_from_regex_rule(rule: str) -> Optional[str]:
-    """Gets the pattern from the regex rule
-
-    Arguments:
-        rule: The full regex rule
-
-    Returns:
-        If the module parameter is search or match, and the pattern parameter exists
-          the pattern is returned
-        Otherwise None
-    """
-    parameters = rule.split(" ")
-    if len(parameters) != 3:
-        return None
-    _, module, pattern = parameters
-    # Do not translate other modules
-    if module not in [item.value for item in RegexModule]:
-        return None
-    # Match is just search but only at the beginning of the string
-    if module == RegexModule.MATCH.value and not pattern.startswith("^"):
-        return f"^{pattern}"
-    return pattern
-
-
-def _get_type_rule_from_rule_list(rule_list: list[str]) -> Optional[str]:
-    """
-    Returns the type rule from a list of rules if there is only one
-    Returns None if there are no type rules
+    if validation_rules:
+        validation_rules = filter_unused_inputted_rules(validation_rules)
+        check_for_duplicate_inputted_rules(validation_rules)
+        check_for_conflicting_inputted_rules(validation_rules)
 
-    Arguments:
-        rule_list: A list of validation rules
+        if get_rule_from_inputted_rules(ValidationRuleName.LIST, validation_rules):
+            js_is_array = True
 
-    Raises:
-        ValueError: When more than one type rule is found
+        js_type = get_js_type_from_inputted_rules(validation_rules)
```

Comment:
I found the logic here difficult to follow.  Why `get_js_type_from_inputted_rules` is not returning the correct `js_type` in the beginning and we have to override it later? if a rule has both date and regex, it seems like js_type will get overwritten first by `if date_rule` and then by `if regex_rule`

Reply:
I also think maybe the mapping between json schema type and validation rules can be more explicit? Just by reading the code here, it is hard to process what the mapping is.